src/dos/dos_keyboard_layout.cpp

```diff
@@ -1128,128 +1167,368 @@ Bitu keyboard_layout::switch_keyboard_layout(const char* new_layout, keyboard_la
+	{"rx",     Country::Russia         }, // Extended Standard
+	{"rx443",  Country::Russia         }, // Extended Typewriter
+	{"sd",     Country::Switzerland    }, // German
+	{"sf",     Country::Switzerland    }, // French
+	{"sg",     Country::Switzerland    }, // German
+	{"si",     Country::Slovenia       },
+	{"sk",     Country::Czech_Slovak   }, // Slovakia
+	{"sp",     Country::Spain          },
+	{"sq",     Country::Albania        }, // No-deadkeys
+	{"sq448",  Country::Albania        }, // Deadkeys
+	{"sr",     Country::Serbia         }, // Deadkey
+	{"su",     Country::Finland        },
+	{"sv",     Country::Sweden         },
+	{"sx",     Country::Spain          }, // International
+	{"tj",     Country::Tadjikistan    },
+	{"tm",     Country::Turkmenistan   },
+	{"tr",     Country::Turkey         }, // QWERTY
+	{"tr440",  Country::Turkey         }, // Non-standard
+	{"tt",     Country::Russia         }, // Tatarstan Standard
+	{"tt443",  Country::Russia         }, // Tatarstan Typewriter
+	{"ua",     Country::Ukraine        }, // 101-key
+	{"uk",     Country::United_Kingdom }, // Standard
+	{"uk168",  Country::United_Kingdom }, // Allternate
+	{"ur",     Country::Ukraine        }, // 101-key
+	{"ur465",  Country::Ukraine        }, // 101-key
+	{"ur1996", Country::Ukraine        }, // 101-key
+	{"ur2001", Country::Ukraine        }, // 102-key
+	{"ur2007", Country::Ukraine        }, // 102-key
+	{"us",     Country::United_States  }, // Standard
+	{"ux",     Country::United_States  }, // International
+	{"uz",     Country::Uzbekistan     },
+	{"vi",     Country::Vietnam        },
+	{"yc",     Country::Serbia         }, // Deadkey
+	{"yc450",  Country::Serbia         }, // No-deadkey
+	{"yu",     Country::Yugoslavia     },
+        // clang-format on
+};
+
+static bool country_number_exists(const int requested_number)
+{
+	for ([[maybe_unused]] const auto &[code, number] : country_code_map)
+		if (requested_number == static_cast<int>(number))
+			return true;
+	return false;
+}
+
+static Country lookup_country_from_code(const char *country_code)
+{
+	if (country_code) {
+		const auto it = country_code_map.find(country_code);
+		if (it != country_code_map.end()) {
+			return it->second;
+		}
+	}
+	return Country::United_States;
+}
+
+constexpr uint16_t assert_codepage(const uint16_t codepage)
+{
+	assert(get_cpx_file_id(codepage) != -1);
+	return codepage;
+}
+
+[[maybe_unused]] constexpr uint16_t lookup_codepage_from_country(const Country country)
+{
+	switch (country) {
+		// grouped in ascending ordered by codepage value
+	case Country::Poland: return assert_codepage(668);
+
+	case Country::Lithuania: return assert_codepage(774);
+
+	case Country::Ukraine: return assert_codepage(848);
+
+	case Country::Latin_America: return assert_codepage(850);
+
+	case Country::Bosnia:
+	case Country::Romania:
+	case Country::Serbia:
+	case Country::Slovenia:
+	case Country::Yugoslavia: return assert_codepage(852);
```

Comment:
Consider using code page 113 (https://en.wikipedia.org/wiki/Code_page#Code_pages_from_other_vendors) for the following countries:

Yugoslavia
Serbia
Montenegro
Croatia
Slovenia
Bosnia
Macedonia

Reply:
Thank you for these finds, @FeralChild64.

This leaves Romania as the only user of code page 852!